src/streamingCalls/streaming.ts

```diff
@@ -119,6 +129,19 @@ export class StreamProxy extends duplexify implements GRPCCallResult {
       stream.on(event, this.emit.bind(this, event));
     });
 
+    // gRPC guaranteed emit the 'status' event but not 'metadata'. And 'status' is the last event to be emit.
+    // Emit the 'response' event if stream has no 'metadata' event.
+    // This avoids stream swallow the other events, such as 'end'.
+    stream.on('status', () => {
+      if (!this._responseHasSent) {
```

Comment:
Do we set `_this.responseHasSent` as we start to write the response, or after the response has finished?

If we set it at the end, this could be a race condition. If we set it at the beginning, this seems like it will work well to me.

Reply:
I think it doesn't matter. The stream event sequence is `metadata` first, then `status` is the last event to be emit.
We only care when the status received, we don't emit the response twice.